### PR TITLE
105:: JPEG & PNG file formats supported (Common)

### PR DESCRIPTION
**Pull request by agnieszkazbikowska-madetech**

### CODE REVIEW

**Comment by agnieszkazbikowska-madetech** [comment]
[JPEG & PNG file formats supported (Common)](https://trello.com/c/W1VVOoQV/105-jpeg-png-file-formats-supported-common)

File: compoments/report-repair/repair-image-upload.js [hunk 1]
```diff
@@ -5,6 +5,34 @@ import Button from '../button';
 import {serviceName} from '../../helpers/constants';
 import imageToBase64 from 'image-to-base64/browser';
 import {isMvpReleaseVersion} from '../../helpers/features';
+import { imageValidator } from '../../helpers/validators';
+
+export const ImagePreview =({image, onDelete}) => {
```
Comment (markWarr) [inline compoments/report-repair/repair-image-upload.js]: I like the way you've exported ImagePreview here so we don't duplicate it in the other page.

File: tests/unit/components/repairImageUpload.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+describe('ImagePreview', () => {
+  test('displays an image', () => {
+    expect(container.querySelector('img').getAttribute('src')).toBe(mockImageBlob);
+  })
+
+  test('displays a delete button & calls the correct function on click', () => {
+    const deleteButton = container.querySelector('button');
+    expect(deleteButton.textContent).toBe('Delete');
+    Simulate.click(deleteButton)
```
Comment (Bhavesh-Patel) [inline tests/unit/components/repairImageUpload.test.js:36]: This test is testing 2 things - the presence of a button and the correct function on click.

Separating to 2 tests would be better to ensure each failure is highlight in a single test run.

File: compoments/report-repair/repair-image-upload.js (hunk 1 above)
Comment (Bhavesh-Patel) [inline compoments/report-repair/repair-image-upload.js:10]: As this component is used within multiple other component (here and [repair-description](https://github.com/Newarkandsherwood/housing-repairs-online-frontend/pull/43/files#diff-b631990ce338ca717f30ed135c4fa88ab9126e3d14d959fc3714ec541cf913aaR8)) , extract this component to a separate file.

As it is _generic_ (i.e. not specific to report-repair), it should be located in the components folder.

**Review by Bhavesh-Patel — CHANGES_REQUESTED** [review]

File: compoments/report-repair/repair-image-upload.js (hunk 1 above)
Comment (agnieszkazbikowska-madetech) [reply compoments/report-repair/repair-image-upload.js:10]: That makes sense. All done!

**Review by markWarr — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -10,9 +10,11 @@
     "test": "jest tests/unit",
     "lint:fix": "eslint --fix */**/*.js tests/**/*.js",
     "cy:open": "cypress open --env CYPRESS_RELEASE_VERSION=full",
+    "cy:openmvp": "cypress open --env CYPRESS_RELEASE_VERSION=mvp",
     "cy:run": "cypress run --env CYPRESS_RELEASE_VERSION=full",
     "cy:runmvp": "cypress run --env CYPRESS_RELEASE_VERSION=mvp  --spec 'tests/cypress/integration/reportRepair/*Mvp.spec.js'",
     "test:integration": "start-server-and-test dev http://localhost:3000 cy:open",
+    "test:integration:mvp": "start-server-and-test start http://localhost:3000 cy:openmvp",
```
Comment (markWarr) [inline package.json:17]: nitpick - should we use dev instead of start here?

**Review by markWarr — APPROVED** [review]

File: compoments/report-repair/repair-image-upload.js (hunk 1 above)
Comment (agnieszkazbikowska-madetech) [reply compoments/report-repair/repair-image-upload.js]: Thanks! I've made one more change & moved it to a separate component as suggested by Bhavesh.

File: package.json (hunk 3 above)
Comment (agnieszkazbikowska-madetech) [reply package.json:17]: This is a really good question! Tbh I always change it to `dev` when I tun the tests locally. @Bhavesh-Patel what do you think?

Comment (Bhavesh-Patel) [reply package.json:17]: Yes, makes senses to change it to `dev`.

Comment (agnieszkazbikowska-madetech) [reply package.json]: I've replaced it.

**Review by Bhavesh-Patel — APPROVED** [review]

**Pull request merged**